### PR TITLE
Add some guidance about communication channels.

### PR DESCRIPTION
**Pull request by geeknoid**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -7,12 +7,38 @@ repositories will go through.  All changes, regardless of whether they are from
...
+- If a design is substantial enough, it deserves a design document. Design documents are written with Google Docs 
+and put into our [Team Drive](https://drive.google.com/corp/drive/u/0/folders/0AIS5p3eW9BCtUk9PVA). If you add a doc
+in the team drive, drop a note in istio-dev to let people know about it.
+
+- Slack is great for quick interactions that demand immediate attention. "Help, XXX doesn't work anymore" or
+"my cluster is on fire, how do I put it out?". Slack is not really appropriate for protracted design discussions which
+usually benefit from more deliberate and thoughtful interactions possible in email.
+
+Keep in mind that Slack can be highly disruptive to the large number of users monitoring the channels. As such,
+please keep funny threads, jokes, memes, etc, out of the main channels. Try to quickly move discussions into
```
Comment (louiscryan) [inline CONTRIBUTING.md]: Maybe point folks to a 'misc' for idle musings, don't want to sound too grinchy :)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -7,12 +7,38 @@ repositories will go through.  All changes, regardless of whether they are from
...
+private chats when appropriate. And move any real design discussions to the mailing lists or to design docs.
```
Comment (louiscryan) [inline CONTRIBUTING.md]: or Github issues

**Comment by louiscryan** [comment]
LGTM - nits

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -7,12 +7,38 @@ repositories will go through.  All changes, regardless of whether they are from
...
+The Istio community has a number of different communication channels. It can be hard to figure out
+where's the right place to ask a question or propose a design. It can be complicated on the receiving end too,
+trying to keep track of all the incoming input. So we'd like to encourage the following:
+
+- Initial design discussions should happen on the [istio-dev](istio-dev@googlegroups.com) mailing list. This is a great
+place to float ideas and get thoughtful feedback. It's archived for everybody to consume into the future.
+
+- If a design is substantial enough, it deserves a design document. Design documents are written with Google Docs 
+and put into our [Team Drive](https://drive.google.com/corp/drive/u/0/folders/0AIS5p3eW9BCtUk9PVA). If you add a doc
+in the team drive, drop a note in istio-dev to let people know about it.
```
Comment (rshriram) [inline CONTRIBUTING.md]: For the moment, can we remove this link ? We don't want it to be wide open. We have no idea who or what is asking for access. And most folks there have r/w perms.

Comment (rshriram) [inline CONTRIBUTING.md]: May be say, we will create a google doc on our shared drive and start a more interactive conversation.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -7,12 +7,38 @@ repositories will go through.  All changes, regardless of whether they are from
...
+
+- Slack is great for quick interactions that demand immediate attention. "Help, XXX doesn't work anymore" or
```
Comment (rshriram) [inline CONTRIBUTING.md]: Let's be explicit here that slack is not for helping end users. From that point of view, I would remove the subsequent quotes (as it's indicating that folks can come there and ask for general help).

**Pull request merged**